src/lib.rs

```diff
@@ -57,6 +57,14 @@ where
     } else {
         let mut ita = path.components();
         let mut itb = base.components();
+
+        // ./foo and foo are the same
+        if let Some(Component::CurDir) = ita.clone().next() {
+            ita.next();
+        }
+        if let Some(Component::CurDir) = itb.clone().next() {
+            itb.next();
+        }
```

Comment:
Should this be in the loop?  what about `"foo/./bar"` and `"foo/bar"`

Maybe even in a while loop?
*   `"././foo"` and `"foo"`  
*  or `"foo/././bar" and `"foo/bar"`.